server/base/src/main/java/org/apache/accumulo/server/tables/TableManager.java

```diff
@@ -286,11 +285,13 @@ public void accept(WatchedEvent event) {
         case NodeCreated:
         case NodeDataChanged:
           // state transition
-          TableState tState = updateTableStateCache(tableId);
-          log.debug("State transition to {} @ {}", tState, event);
-          synchronized (observers) {
-            for (TableObserver to : observers) {
-              to.stateChanged(tableId, tState);
+          if (tableId != null) {
+            TableState tState = updateTableStateCache(tableId);
+            log.debug("State transition to {} @ {}", tState, event);
+            synchronized (observers) {
+              for (TableObserver to : observers) {
+                to.stateChanged(tableId, tState);
+              }
             }
           }
```

Comment:
Is this always going to be a table state change? Or could it be a different kind of watched event on that table path? This code seems to assume that it's always a state change, but I'm not sure if that's true anymore.